pkg/controller/build/build_controller.go

```diff
@@ -98,8 +101,10 @@ func (r *ReconcileBuild) Reconcile(request reconcile.Request) (reconcile.Result,
 			reqLogger.Error(err, "failed validating the ouput secret", "Build", b.Name)
 			b.Status.Reason = err.Error()
 			updateErr := r.client.Status().Update(context.Background(), b)
-			return reconcile.Result{}, fmt.Errorf("errors: %v %v", err, updateErr)
-
+			if updateErr != nil {
```

Comment:
I think we can also discuss here.

If we write the reconcile code like this `return reconcile.Result{}, fmt.Errorf("errors: %v %v", err, updateErr)`

The error messages will report like this:
- validate the output secret error but update status is correct, the error message is `errors: unknown strategy ClusterBuildStrategy2 <nil>`

- validate the output secret error and update statis also fail, the error message is `errors: unknown strategy ClusterBuildStrategy2 Operation cannot be fulfilled on builds.build.dev \"kaniko-golang-build\": StorageError: invalid object, Code: 4, Key: /registry/build.dev/builds/default/kaniko-golang-build, ResourceVersion: 0, AdditionalErrorMsg: Precondition failed: UID in precondition: 6ddb989c-74a2-41b9-a3e9-c54c096f4385, UID in object meta: `

I am not sure if the admin or operator can understand the error clearly by combining the two error messages together?
- What is mean unknown strategy ClusterBuildStrategy2 **nil** ? Does the strategy is nil or something is nil?
- What is mean `unknown strategy ClusterBuildStrategy2 Operation cannot be fulfilled`? Does ClusterBuildStrategy2 exist but some configuration is wrong?

I think the combined message makes admin/operator confuse.

So I think it is better that we just show one error at one time, and our logic will reconcile again, to check the logic or update status next time, if the problem solved, it won't report error anymore. Or continue reporting the related error.

There is JUST ONLY one scenario is not good: if both `validate the output secret error` and `update status error`, the second error will just be shown and doesn't show the first error at this time, but I think it won't happen too often.
But more important is showing each message clearly can avoid confusion.

But yes, we can discuss here.

Reply:
Just the action of omitting errors  is not good. We need to return all of them.  You can always `sugar-coat` the error description to make it more understandable. Error handling is important, not returning them is bad.

Until we do not have a better implementation for handling multiple errors, we should stick with the current approach, where one error can be `nil`. Later we will improve this, with something like https://play.golang.org/p/GEvLz8Xv4Dq (just to give you an idea) 

I´m not resolving this comment yet, while your code is changing the way we deal with multiple errors by ignoring one of them, which is something I would like to avoid. This is my opinion, keeping this open for a broader discussion. @otaviof fyi

Reply:
I had a discussion with @qu1queee ,

We all agree that let us keep two errors (validate error and update error) for reconciliation, like this code style: `fmt.Errorf("errors: %v %v", err, updateErr)`

In future, let us use another PR to refine the format or message for multiple errors.

So I rollback to use this style: `fmt.Errorf("errors: %v %v", err, updateErr)` for build controller.